libs/shared/contentful/src/lib/queries/purchase-with-details-offering-content/types.ts

```diff
@@ -0,0 +1,46 @@
+export interface PurchaseDetailsResult {
+  details: string;
+  productName: string;
+  subtitle: string | null;
+  webIcon: string;
+}
+
+export interface PurchaseDetailsTransformed
+  extends Omit<PurchaseDetailsResult, 'details'> {
+  details: string[];
+}
+
+export interface OfferingCommonContentResult {
+  privacyNoticeUrl: string;
+  privacyNoticeDownloadUrl: string;
+  termsOfServiceUrl: string;
+  termsOfServiceDownloadUrl: string;
+  cancellationUrl: string | null;
+  emailIcon: string | null;
+  successActionButtonUrl: string;
+  successActionButtonLabel: string;
+  newsletterLabelTextCode: string;
+  newsletterSlug: string[];
+}
+
+export interface PurchaseOfferingResult {
+  stripeProductId: string;
+  commonContent: OfferingCommonContentResult;
+}
+
+export interface PurchaseWithDetailsOfferingContentResult {
+  stripePlanChoices: string[];
+  purchaseDetails: PurchaseDetailsResult;
+  offering: PurchaseOfferingResult;
+}
+
+export interface PurchaseWithDetailsOfferingContentTransformed
+  extends Omit<PurchaseWithDetailsOfferingContentResult, 'purchaseDetails'> {
+  purchaseDetails: PurchaseDetailsTransformed;
+}
+
+export interface PurchaseWithDetailsOfferingContentByPlanIdsResult {
```

Comment:
These names feel a little unwieldy. Should we rename it to something simpler, or does the long name provide useful context?

Reply:
I'll defer to the team.

Reply:
Will leave it to a polish PR. :)